app/src/main/java/emu/skyline/settings/SettingsActivity.kt

```diff
@@ -102,6 +109,69 @@ class SettingsActivity : AppCompatActivity() {
             .commit()
     }
 
+    override fun onCreateOptionsMenu(menu : Menu?) : Boolean {
+        (supportFragmentManager.findFragmentById(R.id.settings) as PreferenceFragmentCompat).preferenceScreen.forEach { preferenceCategory ->
+            var noKey = true
+            (preferenceCategory as PreferenceCategory).forEach { preference ->
+                if (preference.key != null) {
+                    mapKeyTitle[preference.title.toString()] = preference
+                    noKey = false
+                }
+            }
+            if (!noKey)
+                mapKeyTitle[preferenceCategory.title.toString()] = preferenceCategory
+        }
```

Comment:
This should definitely not be done here every time the menu is inflated, but after the preference screen has been created.